### DISCUSSION TITLE
'FeedType' not found error for POST_FLAT_FILE_PRICEANDQUANTITYONLY_UPDATE_DATA

### DISCUSSION THREAD

**Discussion by JeffMash** [amzn/selling-partner-api-models#3483] [General] [answered]
I am using the jlevers selling-partner-api.

I'm trying to UPDATE a price on an existing listing by submitting a feed type (POST_FLAT_FILE_PRICEANDQUANTITYONLY_UPDATE_DATA): https://developer-docs.amazon.com/sp-api/docs/feed-type-values

I am submitting the data as an XML Feed but it's throwing the following error:

`PHP Fatal error:  Uncaught Error: Class 'FeedType' not found`

Now, I do see that this FeedType is defined in **selling-partner-api / lib / FeedType.php**, but in there, it says:

```
'contentType' => ContentType::TAB,
        'name' => 'POST_FLAT_FILE_PRICEANDQUANTITYONLY_UPDATE_DATA'
```

**So does this mean that we cannot send an XML to it (because it says TAB and not XML)?**

This is what my XML feed looks like:

```
<?xml version="1.0" encoding="UTF-8"?>
<AmazonEnvelope xsi:noNamespaceSchemaLocation="amzn-envelope.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <Header>
        <DocumentVersion>1.01</DocumentVersion>
        <MerchantIdentifier>A2ZWOLFOFDPJL1</MerchantIdentifier>
    </Header>
    <MessageType>Price</MessageType>    <Message>
                <MessageID>1</MessageID>
                <Price>
                        <SKU>21</SKU>
                        <StandardPrice
currency="USD">48.23</StandardPrice>
                </Price>
        </Message> 
</AmazonEnvelope>
```

**And this is what my code looks like:**

```
try{
	$feedType = FeedType::POST_FLAT_FILE_PRICEANDQUANTITYONLY_UPDATE_DATA;
	$feedsApi = new FeedsApi($config);

	// Create feed document
	$createFeedDocSpec = new Feeds\CreateFeedDocumentSpecification(['content_type' => $feedType['contentType']]);
	$feedDocumentInfo = $feedsApi->createFeedDocument($createFeedDocSpec);
	$feedDocumentId = $feedDocumentInfo->getFeedDocumentId();

	$docToUpload = new SellingPartnerApi\Document($feedDocumentInfo, $feedType);
	$docToUpload->upload($feed); // use feedContents if we want to pass the physical XML file

	$createFeedSpec = new Feeds\CreateFeedSpecification();
	$createFeedSpec->setMarketplaceIds(['ATVPDKIKX0DER']);
	$createFeedSpec->setInputFeedDocumentId($feedDocumentId);
	$createFeedSpec->setFeedType($feedType['name']);

	$createFeedResult = $feedsApi->createFeed($createFeedSpec);
	$feedId = $createFeedResult->getFeedId();
} catch (Exception $e) {
    echo 'Exception when calling SellersApi->getMarketplaceParticipations: ', $e->getMessage(), PHP_EOL;
}
```

**Comment by ShaneOH** [comment, marked as answer]
@JeffMash Are you properly requiring the `FeedType` class in the top of your script? If you're using namespacing you should have `use SellingPartnerApi\FeedType` up top. 

Meanwhile, yes as the name implies, you should be submitting a tab-delimited file with this feed type, not an XML file.

**Reply by JeffMash** [reply]
OMG, I am so embarrassed!  Yep, that was exactly it!   I overlooked that statement at the top of the file!  Thanks so much Shane!

**Reply by ShaneOH** [reply]
You're welcome! And no worries, easy to spot for me since I've done it plenty of times before too 😁
